go/test/endtoend/recovery/shardedpitr/shardedpitr_test.go

```diff
@@ -0,0 +1,621 @@
+
+	for counter := 7; counter <= 10; counter++ {
+		insertRow(t, counter, fmt.Sprintf("prd-%d", counter), false)
+	}
+
+	// wait till replica catchup
+	cluster.VerifyRowsInTabletForTable(t, shard0Replica, keyspaceName, 6, "product")
+	cluster.VerifyRowsInTabletForTable(t, shard1Replica, keyspaceName, 4, "product")
+
+	//start the binlog server and point it to shard0master
+	bs0, err := newBinlogServer(hostname, clusterInstance.GetAndReservePort())
+	defer bs0.stop()
+	require.NoError(t, err)
+
+	err = bs0.start(mysqlMaster{
+		hostname: binlogHost,
+		port:     shard0Master.MysqlctlProcess.MySQLPort,
+		username: mysqlUserName,
+	})
+	require.NoError(t, err)
+
+	//start the binlog server and point it to shard1master
+	bs1, err := newBinlogServer(hostname, clusterInstance.GetAndReservePort())
+	defer bs1.stop()
+	require.NoError(t, err)
+
+	err = bs1.start(mysqlMaster{
+		hostname: binlogHost,
+		port:     shard1Master.MysqlctlProcess.MySQLPort,
+		username: mysqlUserName,
+	})
+	require.NoError(t, err)
+
+	// take the backup (to simulate the regular backup)
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("Backup", shard0Replica.Alias)
+	require.NoError(t, err)
+	// take the backup (to simulate the regular backup)
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("Backup", shard1Replica.Alias)
+	require.NoError(t, err)
+
+	backups, err := clusterInstance.ListBackups(keyspaceName + "/-80")
+	require.NoError(t, err)
+	require.Equal(t, len(backups), 1)
+
+	backups, err = clusterInstance.ListBackups(keyspaceName + "/80-")
+	require.NoError(t, err)
+	require.Equal(t, len(backups), 1)
+
+	// now insert some more data to simulate the changes after regular backup
+	// every insert has some time lag/difference to simulate the time gap between rows
+	// and when we recover to certain time, this time gap will be able to identify the exact eligible row
+	// var timeToRecover string
+	for counter := 11; counter <= 15; counter++ {
+		if counter == 13 { // we want to recovery till this, so noting the time
+			tm := time.Now().Add(1 * time.Second).UTC()
+			timeToRecover = tm.Format(time.RFC3339)
+		}
+		insertRow(t, counter, fmt.Sprintf("prd-%d", counter), true)
+	}
+	currentTime := time.Now().UTC()
+	timeForCompleteBinlogRecover := currentTime.Format(time.RFC3339)
+	_ = currentTime.Format(time.RFC3339)
+
+	// create restoreSnapshot
+	createRestoreSnapshot(t, timeToRecover, partialRestoreKSName)
+
+	// test the recovery with smaller binlog_lookup_timeout
+	shard0recoveryTabletWithSmallTimeout := clusterInstance.NewVttabletInstance("replica", 0, cell)
+	launchRecoveryTablet(t, shard0recoveryTabletWithSmallTimeout, bs0, "1ms", partialRestoreKSName, "-80")
+
+	// test the recovery with smaller binlog_lookup_timeout
+	shard1recoveryTabletWithSmallTimeout := clusterInstance.NewVttabletInstance("replica", 0, cell)
+	launchRecoveryTablet(t, shard1recoveryTabletWithSmallTimeout, bs1, "1ms", partialRestoreKSName, "80-")
```

Comment:
we can remove these, asserting with any of shard with small timeout is fine,

Reply:
Removed